demos/python_demos/text_detection_demo/text_detection_demo.py

```diff
@@ -0,0 +1,264 @@
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import cv2
+import numpy as np
+import argparse
+import sys
+
+
+class PixelLinkDecoder():
+    """ Decoder for Intel's version of PixelLink "text-detection-0003" or "text-detection-0004".
+
+        You will need OpenCV compiled with Inference Engine to use this.
+
+        ** Example of usage:**
+        .. code-block:: python
+
+            td = cv2.dnn.readNet('./text-detection-0003.xml',
+                                 './text-detection-0003.bin')
+            img = cv2.imread('tmp.jpg')
+            blob = cv2.dnn.blobFromImage(img, 1, (1280,768))
+            td.setInput(blob)
+            a, b = td.forward(td.getUnconnectedOutLayersNames())
+            dcd = PixelLinkDecoder()
+            dcd.load(img, a, b)
+            dcd.decode()  # results are in dcd.bboxes
+            dcd.plot_result_pyplot(img)
+    """
+    def __init__(self):
+        pass
+
+    def load(self, image: np.ndarray, pixel_scores: np.ndarray,
+             link_scores: np.ndarray, pixel_conf_threshold=0.8,
+             link_conf_threshold=0.8, four_neighbours=False):
+        """ Load data to decoder
+
+            :param image: BGR image
+            :param pixel_scores: first output of text-detection-0003 or text-detection-0004 model
+            :param link_scores: second output of text-detection-0003 or text-detection-0004 model
+            :param float pixel_conf_threshold: threshold value for pixels
+            :param float link_conf_threshold: threshold value for links
+            :param bool four_neighbours: use 4 or 8 neighbours comparsion algorythm
+        """
+        self.image_shape = image.shape[0:2]
+        self.pixel_scores = self._set_pixel_scores(pixel_scores)
+        self.link_scores = self._set_link_scores(link_scores)
+
+        if four_neighbours:
+            self._get_neighbours = self._get_neighbours_4
+        else:
+            self._get_neighbours = self._get_neighbours_8
+
+        self.pixel_conf_threshold = pixel_conf_threshold
+        self.link_conf_threshold = link_conf_threshold
+        self.pixel_mask = self.pixel_scores >= self.pixel_conf_threshold
+        self.link_mask = self.link_scores >= self.link_conf_threshold
+        self.points = list(zip(*np.where(self.pixel_mask)))
+        self.h, self.w = np.shape(self.pixel_mask)
+        self.group_mask = dict.fromkeys(self.points, -1)
+        self.bboxes = None
+        self.root_map = None
+        self.mask = None
+
+    def _softmax(self, x, axis=None):
+        return np.exp(x - self._logsumexp(x, axis=axis, keepdims=True))
+
+    def _logsumexp(self, a, axis=None, keepdims=False):
```

Comment:
You removed the unused parameters, but they're still referenced in the code. You need to remove the references too.